inc/search.class.php

```diff
@@ -1154,14 +1154,11 @@ static function constructAdditionalSqlForMetacriteria($criteria = [],
                                            $sopt["joinparams"]);
          }
 
-         if (!in_array($sopt["table"]."_".$criterion['itemtype'], $already_link_tables)) {
```

Comment:
This check is invalid, as it does not includes the complexjoin part, and the linkfield part.
Similar check is aleady done in `addLeftJoin`, so it can be removed.